paddle/optimizer/adagrad_optimizer.cc

```diff
@@ -0,0 +1,28 @@
+#include <cmath>
+
+#include "adagrad_optimizer.h"
+
+namespace paddle {
+namespace optimizer {
+
+void AdagradOptimizer::set_weight(Tensor* p) {
+  size_t size = p->size();
+  real* gptr = new real[size];
+  accum_gradient = new Tensor(gptr, size);
+}
+
+void AdagradOptimizer::update(const Tensor* gradient) {
```

Comment:
We need to follow Google C++ code style for C++ function names: https://google.github.io/styleguide/cppguide.html#Function_Names